app/presenters/decidim/proposals/proposal_presenter.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Decidim
```

Comment:
Feedback: I'm not really sure if duplicating this file for the one function is an improvement. I think if we could find an approach to extending Decidim without the code duplication, that could set the software up for the future.

I know we have been relying on this pattern, but imo this is a good time to pause and evaluate since we have the site up and have added some basic functionality.

Question: Is there another way to locate this code that doesn't involve the duplicating strategy? Maybe we should invest in creating our own decidim module just to locate all of our overrides in one spot? Maybe we could add [a helper like Mike suggested](https://github.com/substantial/decidim-seattle/pull/21#discussion_r459804447)?

Question: Is there a way to share this simple algorithm with `equity_quintile` in `app/cells/decidim/tags_cell.rb`.

Reply:
I think I was falling back on my old experience working with rails engines (over a decade ago) where there wasn't so many options available and you just went, "well, it's gross, but this is how it works."

Reading over the modern engines docs, looks like the recommended way these days is to create a concern: https://guides.rubyonrails.org/engines.html#reopening-existing-classes-using-activesupport-concern

How about we try that?

It would be something like:

```
# app/presenters/decidim/proposals/proposal_presenter.rb
module Decidim
  class ProposalPresenter < SimpleDelegator
      include EquityQuintilePresenterFixes
  end
end

# app/presenters/concerns/equity_quintile_presenter_fixes.rb
module EquityQuintilePresenterFixes
  extend ActiveSupport::Concern

  included do
    def equity_quintile
      (@proposal.equity_composite_index_percentile.round(2) * 5).floor() + 1
    end
  end
end
```

Reply:
The other option of making our own engine is a possibility as well. It is kind of appealing to put all our equity changes all in one place: migrations, model changes, import code, etc. It would be a separate gem though (decidim-proposal-equity), which adds some friction, makes it a little harder to iterate on, but we did have a lot of success with that approach on our last project. This is a bigger lift though than introducing a concern.

Reply:
It's a good point that we don't have a ton of time left. I'm concerned about handing off this pattern as the "way to do things" and if we can figure out something lighter weight then that would set up the future devs for success, and they will probably have months of work. So we can view it as an accelerator or improvement for them.

I wonder if there is a way to have our changes live in a separate module within the same repo, as opposed to a separate gem. i.e. a way of organizing code and separating some concerns. Not really sure if that's even possible. But the problem I see is that there are a bunch of files to have the site up and running in the `decidim-seattle` repo, and then on top of that we are layering in our file overrides. So it is hard to see what is part of the decidim site code and what constitutes the changes we've made. (Maybe there is a a pattern or way of delineating between the two that we can document?)

Reply:
I think the big maintenance problem with file overrides is that they will grow out of date when Decidim is updated, so it adds a tax to upgrading the version Decidim, and the more the pattern is used it increases the magnitude of that tax. So, when some security issue arises that requires a future team to upgrade Decidim, they have to manually resolve our changes into new files.

Because of that, imo we should consider avoiding the pattern where possible and if we can actually set up a better foundation for hand-off, that would be awesome. I feel that as we wrap up the "equity piece" in its current design, refactoring is a good next step. Otherwise, we maybe we just have to identify this tech debt item as something future devs should prioritize fixing?

Reply:
@johnjensenish We found a solution that removes all the duplication. Check it out.